packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -38,33 +43,98 @@ type DropdownCoreProps = {
 };
 
 export default class DropdownCore extends React.Component<DropdownCoreProps> {
+    node: ?HTMLDivElement;
+    // TODO(sophie): figure out flow typing
+    handleClick: (event: any) => void;
+
     static defaultProps = {
         alignment: "left",
     };
 
+    constructor(props: DropdownCoreProps) {
+        super(props);
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    componentWillMount() {
+        document.addEventListener("click", this.handleClick, false);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener("click", this.handleClick, false);
+    }
+
+    handleClick(event: SyntheticEvent<>) {
```

Comment:
This should be pre-bound since you're passing it directly to `add`/`removeEventListenere`.
```
handleClick = (event: SyntheticMouseEvent<"click">) => { ... }
```